litgpt/utils.py

```diff
@@ -557,13 +558,40 @@ def instantiate_bnb_optimizer(optimizer, model_parameters):
 
 
 def instantiate_torch_optimizer(optimizer, model_parameters, **kwargs):
+    # Special care taken where some optimizers do not have parameter "fused" like:
+    #   bnb.optim.AdamW8bit
+    #   grokadamw.GrokAdamW
+    
     if isinstance(optimizer, str):
-        optimizer_cls = getattr(torch.optim, optimizer)
+        if "." in optimizer:
+            class_module, class_name = optimizer.rsplit(".", 1)
+        else:
+            class_module, class_name = "torch.optim", optimizer
+
+        module = __import__(class_module, fromlist=[class_name])
+        optimizer_cls = getattr(module, class_name)
+
+        if "fused" in kwargs and "fused" not in inspect.signature(optimizer_cls).parameters:
+            kwargs = dict(kwargs)   # copy
+            del kwargs["fused"]
+
         optimizer = optimizer_cls(model_parameters, **kwargs)
-    else:
+    elif isinstance(optimizer, dict):
         optimizer = dict(optimizer)  # copy
+
+        class_module, class_name = optimizer["class_path"].rsplit(".", 1)
+        module = __import__(class_module, fromlist=[class_name])
+        optimizer_cls = getattr(module, class_name)
+
+        if "fused" in kwargs and "fused" not in inspect.signature(optimizer_cls).parameters:
+            kwargs = dict(kwargs)   # copy
+            del kwargs["fused"]
+        
         optimizer["init_args"].update(kwargs)
         optimizer = instantiate_class(model_parameters, optimizer)
```

Comment:
```suggestion
    elif isinstance(optimizer, dict):
        optimizer = dict(optimizer)
        class_module, class_name = optimizer["class_path"].rsplit(".", 1)
        module = __import__(class_module, fromlist=[class_name])
        optimizer_cls = getattr(module, class_name)

        valid_params = set(inspect.signature(optimizer_cls).parameters)
        kwargs = {key: value for key, value in kwargs.items() if key in valid_params}
        
        optimizer["init_args"].update(kwargs)
        optimizer = instantiate_class(model_parameters, optimizer)
```

Similar to above, this would be a more general solution. What do you think?

Reply:
I like your more general approach by filtering params/args to only valid ones, so everything else is rejected.